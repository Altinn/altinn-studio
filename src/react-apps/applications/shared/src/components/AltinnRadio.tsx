import { FormControlLabel, Radio } from '@material-ui/core';
import { createMuiTheme, createStyles, withStyles, WithStyles } from '@material-ui/core/styles';
import * as React from 'react';
import altinnTheme from '../theme/altinnStudioTheme';

export interface IAltinnRadioComponentProvidedProps extends WithStyles<typeof styles> {
  id?: any;
  value?: any;
  checked?: boolean;
  onChange?: any;
  label?: string;
  disabled?: boolean;
}

const theme = createMuiTheme(altinnTheme);

export interface IAltinnRadioComponentState {
}

const styles = () => createStyles({
  altinnFormControlLabel: {
    fontSize: '1.6rem',
    marginRight: '0rem',
  },
  altinnFormControl: {
    marginRight: '6rem',
  },
  altinnRadio: {
    color: theme.altinnPalette.primary.blueDark + '!important',
    paddingTop: '0rem',
    paddingBottom: '0rem',
  },
});

export class AltinnRadio extends
  React.Component<IAltinnRadioComponentProvidedProps, IAltinnRadioComponentState> {

  public render() {
    if (!this.props.label) {
      return (
        <Radio
          classes={{ root: this.props.classes.altinnRadio }}
          id={this.props.id}
          checked={this.props.checked}
          onChange={this.props.onChange}
          disabled={this.props.disabled}
          value={this.props.value}
        />
      );
    } else {
      return (
        <FormControlLabel
          classes={{ root: this.props.classes.altinnFormControl, label: this.props.classes.altinnFormControlLabel }}
          label={this.props.label}
          control={
            <Radio
              classes={{ root: this.props.classes.altinnRadio }}
              id={this.props.id}
              checked={this.props.checked}
              onChange={this.props.onChange}
              disabled={this.props.disabled}
              value={this.props.value}
            />
          }
        />
      );
    }
  }
}

export default withStyles(styles)(AltinnRadio);
