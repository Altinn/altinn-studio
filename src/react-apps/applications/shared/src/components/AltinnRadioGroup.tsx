import { RadioGroup, Typography } from '@material-ui/core';
import { createStyles, withStyles, WithStyles } from '@material-ui/core/styles';
import * as React from 'react';

export interface IAltinnRadioGroupComponentProvidedProps extends WithStyles<typeof styles> {
  id?: any;
  value: any;
  onChange?: any;
  className?: any;
  name?: string;
  row?: boolean;
  description?: string;
}

export interface IAltinnRadioGroupComponentState {
}

const styles = () => createStyles({
  altinnRadioGroup: {
    marginTop: '0rem',
    marginBottom: '0rem',
  },
  altinnRadioGroupWrapper: {
    marginTop: '2.4rem',
  },
  altinnRadioGroupDescription: {
    fontSize: '1.6rem',
  },
});

export class AltinnRadioGroup extends
  React.Component<IAltinnRadioGroupComponentProvidedProps, IAltinnRadioGroupComponentState> {

  public render() {
    return (
      <div className={this.props.classes.altinnRadioGroupWrapper}>
        {this.props.description &&
          <Typography classes={{ root: this.props.classes.altinnRadioGroupDescription }}>
            {this.props.description}
          </Typography>
        }
        <RadioGroup
          classes={{ root: this.props.classes.altinnRadioGroup }}
          onChange={this.props.onChange}
          value={this.props.value}
          className={this.props.className}
          name={this.props.name}
          id={this.props.id}
          row={this.props.row}
        >
          {this.props.children}
        </RadioGroup>
      </div>
    );
  }
}

export default withStyles(styles)(AltinnRadioGroup);
