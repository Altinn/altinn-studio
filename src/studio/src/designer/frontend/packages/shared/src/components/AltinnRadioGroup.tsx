import React from 'react';
import { RadioGroup, Typography } from '@mui/material';
import { makeStyles } from '@mui/styles';

export interface IAltinnRadioGroupProps {
  id?: any;
  value: any;
  onChange?: any;
  className?: any;
  name?: string;
  row?: boolean;
  description?: string;
  children: React.ReactNode;
}

const useStyles = makeStyles({
  altinnRadioGroup: {
    marginTop: '0rem',
    marginBottom: '0rem',
  },
  altinnRadioGroupWrapper: {
    marginTop: '2.4rem',
  },
  altinnRadioGroupDescription: {
    fontSize: '1.6rem',
  },
});

export const AltinnRadioGroup = ({
  description,
  onChange,
  value,
  className,
  name,
  row,
  id,
  children,
}: IAltinnRadioGroupProps) => {
  const classes = useStyles();

  return (
    <div className={classes.altinnRadioGroupWrapper}>
      {description && (
        <Typography classes={{ root: classes.altinnRadioGroupDescription }}>
          {description}
        </Typography>
      )}
      <RadioGroup
        classes={{ root: classes.altinnRadioGroup }}
        onChange={onChange}
        value={value}
        className={className}
        name={name}
        id={id}
        row={row}
      >
        {children}
      </RadioGroup>
    </div>
  );
};

export default AltinnRadioGroup;
