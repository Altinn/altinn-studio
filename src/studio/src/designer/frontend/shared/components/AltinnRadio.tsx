import { FormControlLabel, Radio, makeStyles } from '@material-ui/core';
import { createTheme } from '@material-ui/core/styles';
import React from 'react';
import altinnTheme from '../theme/altinnStudioTheme';

export interface IAltinnRadioProps {
  id?: any;
  value?: any;
  checked?: boolean;
  onChange?: any;
  label?: string;
  disabled?: boolean;
}

const theme = createTheme(altinnTheme);

const useStyles = makeStyles({
  altinnFormControlLabel: {
    fontSize: '1.6rem',
    marginRight: '0rem',
  },
  altinnFormControl: {
    marginRight: '6rem',
  },
  altinnRadio: {
    color: `${theme.altinnPalette.primary.blueDark} !important`,
    paddingTop: '0rem',
    paddingBottom: '0rem',
  },
});

export const AltinnRadio = ({
  label,
  id,
  checked,
  onChange,
  disabled,
  value,
}: IAltinnRadioProps) => {
  const classes = useStyles();

  if (!label) {
    return (
      <Radio
        classes={{ root: classes.altinnRadio }}
        id={id}
        checked={checked}
        onChange={onChange}
        disabled={disabled}
        value={value}
      />
    );
  }

  return (
    <FormControlLabel
      classes={{
        root: classes.altinnFormControl,
        label: classes.altinnFormControlLabel,
      }}
      label={label}
      control={
        <Radio
          classes={{ root: classes.altinnRadio }}
          id={id}
          checked={checked}
          onChange={onChange}
          disabled={disabled}
          value={value}
        />
      }
    />
  );
};

export default AltinnRadio;
