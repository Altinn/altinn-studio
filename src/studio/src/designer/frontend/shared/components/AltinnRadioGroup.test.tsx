import { Typography } from '@material-ui/core';
import { mount } from 'enzyme';
import React from 'react';
import * as renderer from 'react-test-renderer';
import { AltinnRadio } from './AltinnRadio';
import { AltinnRadioGroup } from './AltinnRadioGroup';

describe('AltinnRadioButtonGroup', () => {
  let mockValue: string;
  let mockId: string;
  let mockDescription: string;
  let mockOnChange: any;
  let mockRow: boolean;
  let mockClassName: any;

  beforeEach(() => {
    mockValue = 'mock-value';
    mockId = 'mock-id';
    mockDescription = 'mock-description';
    mockRow = true;
    mockClassName = 'mock-classname';
    mockOnChange = jest.fn();
  });

  it('Should match snapshot with the least amount of params', () => {
    const rendered = renderer.create(
      <AltinnRadioGroup value={mockValue} name='radioGroup'>
        <AltinnRadio />
      </AltinnRadioGroup>,
    );
    expect(rendered).toMatchSnapshot();
  });

  it('Should match snapshot with all params ', () => {
    const rendered = renderer.create(
      <AltinnRadioGroup
        value={mockValue}
        onChange={mockOnChange}
        id={mockId}
        description={mockDescription}
        row={mockRow}
        className={mockClassName}
        name='radioGroup'
      >
        <AltinnRadio />
      </AltinnRadioGroup>,
    );
    expect(rendered).toMatchSnapshot();
  });

  it('Should render children', () => {
    const wrapper = mount(
      <AltinnRadioGroup value={mockValue} name='radioGroup'>
        <AltinnRadio />
        <AltinnRadio />
      </AltinnRadioGroup>,
    );
    expect(wrapper.find(AltinnRadio)).toHaveLength(2);
  });

  it('Should render description typography if description is supplied', () => {
    const wrapper = mount(
      <AltinnRadioGroup
        value={mockValue}
        description={mockDescription}
        name='radioGroup'
      >
        <AltinnRadio />
      </AltinnRadioGroup>,
    );
    expect(wrapper.find(Typography)).toHaveLength(1);
  });
});
