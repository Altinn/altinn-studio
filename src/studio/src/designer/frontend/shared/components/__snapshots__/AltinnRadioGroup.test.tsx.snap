// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AltinnRadioButtonGroup Should match snapshot with all params  1`] = `
<div
  className="makeStyles-altinnRadioGroupWrapper-2"
>
  <p
    className="MuiTypography-root makeStyles-altinnRadioGroupDescription-3 MuiTypography-body1"
  >
    mock-description
  </p>
  <div
    className="MuiFormGroup-root makeStyles-altinnRadioGroup-1 mock-classname MuiFormGroup-row"
    id="mock-id"
    role="radiogroup"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiRadio-root makeStyles-altinnRadio-6 MuiRadio-colorSecondary MuiIconButton-colorSecondary"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input-10"
          name="radioGroup"
          onChange={[Function]}
          type="radio"
        />
        <div
          className="PrivateRadioButtonIcon-root-11"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-12"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </div>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </span>
  </div>
</div>
`;

exports[`AltinnRadioButtonGroup Should match snapshot with the least amount of params 1`] = `
<div
  className="makeStyles-altinnRadioGroupWrapper-2"
>
  <div
    className="MuiFormGroup-root makeStyles-altinnRadioGroup-1"
    role="radiogroup"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiRadio-root makeStyles-altinnRadio-6 MuiRadio-colorSecondary MuiIconButton-colorSecondary"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input-10"
          name="radioGroup"
          onChange={[Function]}
          type="radio"
        />
        <div
          className="PrivateRadioButtonIcon-root-11"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-12"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </div>
      </span>
    </span>
  </div>
</div>
`;
